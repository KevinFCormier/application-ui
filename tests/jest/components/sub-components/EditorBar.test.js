/*******************************************************************************
 * Licensed Materials - Property of IBM
 * 5737-E67
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import { shallow } from 'enzyme'

import EditorBar from '../../../../src-web/components/ApplicationTopologyModule/components/EditorBar'

describe('EditorBar components', () => {
  it('renders as expected', () => {
    const component = shallow(
      <EditorBar
        hasUndo={false}
        hasRedo={false}
        exceptions={[]}
        gotoEditorLine={jest.fn()}
        handleEditorCommand={jest.fn()}
        handleSearchChange={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
