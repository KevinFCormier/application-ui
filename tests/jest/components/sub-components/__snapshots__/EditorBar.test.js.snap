// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorBar components renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditorBar
    exceptions={Array []}
    gotoEditorLine={[MockFunction]}
    handleEditorCommand={[MockFunction]}
    handleSearchChange={[MockFunction]}
    hasRedo={false}
    hasUndo={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          aria-label="Search"
          className="editor-bar-search"
          id="Search"
          role="region"
        >
          <Search
            aria-label="Search"
            id="search-name"
            labelText=""
            light={false}
            onChange={[Function]}
            placeHolderText="Search"
            small={true}
            type="text"
          />
        </div>,
        Array [
          <EditorButton
            button={
              Object {
                "command": "previous",
                "disabled": true,
                "icon": "previous",
                "tooltip": "Previous",
              }
            }
            command="previous"
            handleClick={[Function]}
          />,
          <EditorButton
            button={
              Object {
                "command": "next",
                "disabled": true,
                "icon": "next",
                "tooltip": "Next",
              }
            }
            command="next"
            handleClick={[Function]}
          />,
          <EditorButton
            button={
              Object {
                "command": "spacer1",
                "spacer": true,
              }
            }
            command="spacer1"
            handleClick={[Function]}
          />,
          <EditorButton
            button={
              Object {
                "command": "undo",
                "disabled": true,
                "icon": "undo",
                "tooltip": "Undo",
              }
            }
            command="undo"
            handleClick={[Function]}
          />,
          <EditorButton
            button={
              Object {
                "command": "redo",
                "disabled": true,
                "icon": "redo",
                "tooltip": "Redo",
              }
            }
            command="redo"
            handleClick={[Function]}
          />,
          <EditorButton
            button={
              Object {
                "command": "restore",
                "disabled": true,
                "icon": "restore",
                "tooltip": "Restore",
              }
            }
            command="restore"
            handleClick={[Function]}
          />,
          <EditorButton
            button={
              Object {
                "command": "spacer2",
                "spacer": true,
              }
            }
            command="spacer2"
            handleClick={[Function]}
          />,
          <EditorButton
            button={
              Object {
                "command": "update",
                "disabled": true,
                "icon": "deploy",
                "tooltip": "Update",
              }
            }
            command="update"
            handleClick={[Function]}
          />,
        ],
        false,
      ],
      "className": "editor-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Search",
          "children": <Search
            aria-label="Search"
            id="search-name"
            labelText=""
            light={false}
            onChange={[Function]}
            placeHolderText="Search"
            small={true}
            type="text"
          />,
          "className": "editor-bar-search",
          "id": "Search",
          "role": "region",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Search",
            "id": "search-name",
            "labelText": "",
            "light": false,
            "onChange": [Function],
            "placeHolderText": "Search",
            "small": true,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "previous",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "previous",
            "disabled": true,
            "icon": "previous",
            "tooltip": "Previous",
          },
          "command": "previous",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "next",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "next",
            "disabled": true,
            "icon": "next",
            "tooltip": "Next",
          },
          "command": "next",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "spacer1",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "spacer1",
            "spacer": true,
          },
          "command": "spacer1",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "undo",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "undo",
            "disabled": true,
            "icon": "undo",
            "tooltip": "Undo",
          },
          "command": "undo",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "redo",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "redo",
            "disabled": true,
            "icon": "redo",
            "tooltip": "Redo",
          },
          "command": "redo",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "restore",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "restore",
            "disabled": true,
            "icon": "restore",
            "tooltip": "Restore",
          },
          "command": "restore",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "spacer2",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "spacer2",
            "spacer": true,
          },
          "command": "spacer2",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "update",
        "nodeType": "class",
        "props": Object {
          "button": Object {
            "command": "update",
            "disabled": true,
            "icon": "deploy",
            "tooltip": "Update",
          },
          "command": "update",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            aria-label="Search"
            className="editor-bar-search"
            id="Search"
            role="region"
          >
            <Search
              aria-label="Search"
              id="search-name"
              labelText=""
              light={false}
              onChange={[Function]}
              placeHolderText="Search"
              small={true}
              type="text"
            />
          </div>,
          Array [
            <EditorButton
              button={
                Object {
                  "command": "previous",
                  "disabled": true,
                  "icon": "previous",
                  "tooltip": "Previous",
                }
              }
              command="previous"
              handleClick={[Function]}
            />,
            <EditorButton
              button={
                Object {
                  "command": "next",
                  "disabled": true,
                  "icon": "next",
                  "tooltip": "Next",
                }
              }
              command="next"
              handleClick={[Function]}
            />,
            <EditorButton
              button={
                Object {
                  "command": "spacer1",
                  "spacer": true,
                }
              }
              command="spacer1"
              handleClick={[Function]}
            />,
            <EditorButton
              button={
                Object {
                  "command": "undo",
                  "disabled": true,
                  "icon": "undo",
                  "tooltip": "Undo",
                }
              }
              command="undo"
              handleClick={[Function]}
            />,
            <EditorButton
              button={
                Object {
                  "command": "redo",
                  "disabled": true,
                  "icon": "redo",
                  "tooltip": "Redo",
                }
              }
              command="redo"
              handleClick={[Function]}
            />,
            <EditorButton
              button={
                Object {
                  "command": "restore",
                  "disabled": true,
                  "icon": "restore",
                  "tooltip": "Restore",
                }
              }
              command="restore"
              handleClick={[Function]}
            />,
            <EditorButton
              button={
                Object {
                  "command": "spacer2",
                  "spacer": true,
                }
              }
              command="spacer2"
              handleClick={[Function]}
            />,
            <EditorButton
              button={
                Object {
                  "command": "update",
                  "disabled": true,
                  "icon": "deploy",
                  "tooltip": "Update",
                }
              }
              command="update"
              handleClick={[Function]}
            />,
          ],
          false,
        ],
        "className": "editor-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Search",
            "children": <Search
              aria-label="Search"
              id="search-name"
              labelText=""
              light={false}
              onChange={[Function]}
              placeHolderText="Search"
              small={true}
              type="text"
            />,
            "className": "editor-bar-search",
            "id": "Search",
            "role": "region",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Search",
              "id": "search-name",
              "labelText": "",
              "light": false,
              "onChange": [Function],
              "placeHolderText": "Search",
              "small": true,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "previous",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "previous",
              "disabled": true,
              "icon": "previous",
              "tooltip": "Previous",
            },
            "command": "previous",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "next",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "next",
              "disabled": true,
              "icon": "next",
              "tooltip": "Next",
            },
            "command": "next",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "spacer1",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "spacer1",
              "spacer": true,
            },
            "command": "spacer1",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "undo",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "undo",
              "disabled": true,
              "icon": "undo",
              "tooltip": "Undo",
            },
            "command": "undo",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "redo",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "redo",
              "disabled": true,
              "icon": "redo",
              "tooltip": "Redo",
            },
            "command": "redo",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "restore",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "restore",
              "disabled": true,
              "icon": "restore",
              "tooltip": "Restore",
            },
            "command": "restore",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "spacer2",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "spacer2",
              "spacer": true,
            },
            "command": "spacer2",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "update",
          "nodeType": "class",
          "props": Object {
            "button": Object {
              "command": "update",
              "disabled": true,
              "icon": "deploy",
              "tooltip": "Update",
            },
            "command": "update",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
